api/app/src/routes/medical/link.ts

```diff
@@ -1,95 +1,165 @@
 import { Request, Response } from "express";
 import Router from "express-promise-router";
-import { asyncHandler, getCxIdOrFail, getPatientIdFromQueryOrFail } from "../util";
-const router = Router();
 import status from "http-status";
-import { getPatient } from "../../command/medical/patient/get-patient";
+import { uniqBy } from "lodash";
+
+import {
+  asyncHandler,
+  getCxIdOrFail,
+  getPatientIdFromParamsOrFail,
+  getEntityIdFromBodyOrFail,
+} from "../util";
+const router = Router();
+import { getPatientWithDependencies } from "../../command/medical/patient/get-patient";
+import { updatePatientLinks } from "../../command/medical/patient/update-patient-link";
+import cwCommands from "../../external/commonwell";
+import { PatientLinks } from "./schemas/link";
+import { LinkSource } from "./schemas/link";
+import { driversLicenseURIs } from "../../shared/oid";
+import { Link } from "./schemas/link";
 
 /** ---------------------------------------------------------------------------
- * POST /link
+ * POST /patient/:patientId/link/:source
  *
  * Creates link to the specified entity.
+ * @param   req.params.patientId   Patient ID to link to a person.
+ * @param   req.params.source      HIE from where the link is made too.
+ * @param   req.body.entityId      Person ID to link to the patient.
  *
- * @return  {Link}  The created link.
+ * @return  200
  */
 router.post(
-  "/",
+  "/:patientId/link/:source",
   asyncHandler(async (req: Request, res: Response) => {
     const cxId = getCxIdOrFail(req);
-    const patientId = getPatientIdFromQueryOrFail(req);
-    const patient = await getPatient({ id: patientId, cxId });
-    // get linkSource from query params -> this should be "CommonWell"
-    console.log(patient);
-
-    // CommonWell.patientLink();
-    // store the following data in our DB in the patient:
-    //    - link ID
-    //    - person ID
-    // can put this into a JSONB column `link_data` with structure:
-    //    {["CommonWell"]: {cw_link_id, cw_person_id}}
-
-    return res.status(status.OK).json({});
+    const patientId = getPatientIdFromParamsOrFail(req);
+    const entityId = getEntityIdFromBodyOrFail(req);
+
+    const { patient, organization } = await getPatientWithDependencies({ id: patientId, cxId });
+    const linkSource = req.params.source;
+
+    // TODO: HANDLE OTHER HIE's
+    if (linkSource === LinkSource.commonWell) {
+      await cwCommands.link.create(entityId, patient, organization);
+
+      await updatePatientLinks({
+        id: patientId,
+        cxId,
+        linkData: {
+          ...patient.linkData,
+          [linkSource]: {
+            cw_person_id: entityId,
+          },
+        },
+      });
```

Comment:
If we go with using the patient data to store this, we could use `setCommonwellId()`.

The reason to have a specific command for it, is so our `patient.update()` is not mixed with partner logic.